skops/hub_utils/_hf_hub.py

```diff
@@ -339,6 +349,12 @@ def init(
 
     dst.mkdir(parents=True, exist_ok=True)
 
+    # add intelex requirement, if it's used and not already in requirements
+    if use_intelex and not any(
+        r.startswith("scikit-learn-intelex") for r in requirements
+    ):
+        requirements.append("scikit-learn-intelex")
```

Comment:
should we add the current version as well? I think we should

Reply:
Hmm, maybe. That would require sklearnex to be installed though, and technically, users should be allowed to enable the intelex even if they don't have sklearnex installed.

The second problem is that I can't find how to determine the sklearnex version. What I found in their code is this:

https://github.com/intel/scikit-learn-intelex/blob/6b31cc8499d12a1ccbe8200b0e90993f6731c44e/setup_sklearnex.py#L24

So they use some kind of calendar versioning but I haven't found the version as an attribute on the package.

There's also this function:

https://github.com/intel/scikit-learn-intelex/blob/6b31cc8499d12a1ccbe8200b0e90993f6731c44e/sklearnex/_utils.py#L74

But it doesn't seem to return the version (AFAICT, it requires you to pass the version and does some checks on it).

If users want to have a specific version, they can indicate it, it's just a small extra step.

Reply:
what would `importlib.metadata.version` give you?

Reply:
Hmm

```
$ python -c "from importlib.metadata import version; print(version('scikit-learn-intelex'))"
2023.20221220.91616
$ python -m pip install "scikit-learn-intelex==2023.20221220.91616"
ERROR: Could not find a version that satisfies the requirement scikit-learn-intelex==2023.20221220.91616 (from versions: 2021.2.1, 2021.2.2, 2021.2.3, 2021.3.0, 2021.4.0, 2021.5.1, 2021.5.3, 2021.6.0, 2021.6.2, 2021.6.3, 2023.0.0, 2023.0.1)
ERROR: No matching distribution found for scikit-learn-intelex==2023.20221220.91616
```